From 0bab8aee77cfc2ffdbc6d3ef6a869284bc12dff4 Mon Sep 17 00:00:00 2001
From: Muthu Subramanian <muthu.subramanian.karunanidhi@ericsson.com>
Date: Thu, 23 Oct 2014 20:30:26 +0530
Subject: [PATCH] fdo#85247: copy and paste of a slide results in a blue slide
 in presentation mode.

Seems like "Default" is handled specially or probably
the copy-buffer is broken
This now checks for slide 'type' as well, before cosidering it a duplicate

Change-Id: If8c472bcefb54cee72e7411f92a76b4e2db2b6ce
---
 sd/source/core/drawdoc3.cxx | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sd/source/core/drawdoc3.cxx b/sd/source/core/drawdoc3.cxx
index e379ed0..d3e92f9 100644
--- a/sd/source/core/drawdoc3.cxx
+++ b/sd/source/core/drawdoc3.cxx
@@ -106,9 +106,11 @@ void InsertBookmarkAsPage_FindDuplicateLayouts::operator()( SdDrawDocument& rDoc
 
         OUString aTest(aFullTest);
 
-        if (aTest == aLayout)
+        if (aTest == aLayout && pBMMPage->GetPageKind() == pTestPage->GetPageKind())
         {
-            if( bRenameDuplicates && pTestPage->getHash() != pBMMPage->getHash() )
+            // Ignore Layouts with "Default" these seem to be special - in the sense that there are lot of assumption all over Impress
+            // about this
+            if( bRenameDuplicates && aTest != OUString( SdResId( STR_LAYOUT_DEFAULT_NAME ) ) && pTestPage->getHash() != pBMMPage->getHash() )
             {
                 pBookmarkDoc->RenameLayoutTemplate( pBMMPage->GetLayoutName(), OUString(pBMMPage->GetName())+=OUString("_") );
                 aLayout = pBMMPage->GetName();
-- 
1.9.3

